ipi/engine/forcefields.py

```diff
@@ -1552,6 +1550,12 @@ def calc_dipole_xyz_mm(self, pos, n_bath, charge_array_bath):
         dx_array, dy_array, dz_array = [], [], []
         for idx in range(n_bath):
             pos_bath = pos[ndim_local * idx : ndim_local * (idx + 1)]
+            # check the dimension of charge array
+            if np.size(pos_bath[::3]) != np.size(charge_array_bath):
+                raise ValueError(
```

Comment:
Please use  softexit here:

Something like:

from ipi.utils.softexit import softexit (add this at the beginning of the file)
softexit.trigger("<Error message>")